### PR TITLE
add xgboost online prediction notebook

### PR DESCRIPTION
**Pull request by sirtorry**
some sklearn notebook fixes as well.

### CODE REVIEW

**Comment by nnegrey** [comment]
You should add a link to your sample from the top level README.

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Created ml engine model [projects/demos-191212/models/census].\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud ml-engine models create $MODEL_NAME --regions us-central1"
```
Comment (nnegrey) [inline xgboost/notebooks/Online Prediction with XGBoost.ipynb]: Update to $REGION

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 4: Create a model version\n",
+    "\n",
+    "Now it’s time to get your model online and ready for predictions. The model version requires a few components as specified [here](https://cloud.google.com/ml-engine/reference/rest/v1/projects.models.versions#Version).\n",
+    "\n",
+    "* __name__ - The name specified for the version when it was created. This will be the `VERSION_NAME` variable you declared at the beginning.\n",
+    "* __deployment Uri__ (curl) or __origin__ (gcloud) - The Google Cloud Storage location of the trained model used to create the version. This is the bucket that you uploaded the model to with your `BUCKET_ID`\n",
```
Comment (nnegrey) [inline xgboost/notebooks/Online Prediction with XGBoost.ipynb]: Drop "or __origin__ (gcloud)" since we removed the gcloud step.

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0.09139516949653625]\r\n"
```
Comment (nnegrey) [inline xgboost/notebooks/Online Prediction with XGBoost.ipynb]: Can you add clarification on what the output means? (This one is supposed to be close to zero, since it was someone who makes <50K, right?)

Or 

Should we have it match the Label we showed in our Table? True/False?

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb [hunk 4]
```diff
@@ -0,0 +1,627 @@
...
+   "source": [
+    "! gcloud ml-engine predict --model $MODEL_NAME --version $VERSION_NAME --json-instances more_than_50K.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use Python to make online predictions\n",
+    "We'll test the model with the entire test set and print out some of the results. Also, realise how the cells above return floats instead of booleans. Let's deal with that so the output type of the predictions match those of the test set labels.\n",
```
Comment (nnegrey) [inline xgboost/notebooks/Online Prediction with XGBoost.ipynb]: I'd move your comment "realize how the cells ..." to a note above and say you'll fix it in the next step. Helps clarify why they got a float when maybe they weren't expecting that. Thoughts?

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb [hunk 5]
```diff
@@ -0,0 +1,634 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Prediction: False\tLabel: False\n",
```
Comment (nkashy1) [inline xgboost/notebooks/Online Prediction with XGBoost.ipynb]: This is a little hard to read. Could you run this through `matplotlib` or `seaborn` to produce a confusion matrix or something similar?

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb [hunk 6]
```diff
@@ -0,0 +1,634 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 0: Setup\n",
+    "* [Create a project on GCP](https://cloud.google.com/resource-manager/docs/creating-managing-projects)\n",
+    "* [Create a GCS Bucket](https://cloud.google.com/storage/docs/quickstart-console)\n",
+    "* [Enable Cloud Machine Learning Engine and Compute Engine APIs](https://console.cloud.google.com/flows/enableapi?apiid=ml.googleapis.com,compute_component&_ga=2.217405014.1312742076.1516128282-1417583630.1516128282)\n",
+    "* [Install XGBoost](http://xgboost.readthedocs.io/en/latest/build.html)\n",
```
Comment (nkashy1) [inline xgboost/notebooks/Online Prediction with XGBoost.ipynb]: These steps are good and we don't have the corresponding steps in the scikit-learn notebook?

I've created an issue to track this work: https://github.com/GoogleCloudPlatform/cloudml-samples/issues/163

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb [hunk 7]
```diff
@@ -0,0 +1,634 @@
...
+    "# Prerequisites\n",
+    "Before we begin, let’s cover some of the different tools you’ll use to get online prediction up and running on ML Engine. \n",
+    "\n",
+    "[Google Cloud Platform](https://cloud.google.com/) (GCP) lets you build and host applications and websites, store data, and analyze data on Google's scalable infrastructure.\n",
+    "\n",
+    "[Cloud ML Engine](https://cloud.google.com/ml-engine/) is a managed service that enables you to easily build machine learning models that work on any type of data, of any size.\n",
+    "\n",
+    "[Google Cloud Storage](https://cloud.google.com/storage/) (GCS) is a unified object storage for developers and enterprises, from live data serving to data analytics/ML to data archiving.\n",
+    "\n",
+    "[Cloud SDK](https://cloud.google.com/sdk/) is a set of tools for Cloud Platform. It contains gcloud, gsutil, and bq, which you can use to access Google Compute Engine, Google Cloud Storage, Google BigQuery, and other products and services from the command-line.\n"
```
Comment (nkashy1) [inline xgboost/notebooks/Online Prediction with XGBoost.ipynb]: We should make it more explicit that we assume the Cloud SDK to be installed in the environment that Jupyter is running in - for example, if a user decides to run Jupyter in a Docker container, it is unlikely to have the `gcloud` command available to it and all the `! gcloud ...` cells will fail.

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb (hunk 5 above)
Comment (sirtorry) [reply xgboost/notebooks/Online Prediction with XGBoost.ipynb]: hmm how about i show a confusion matrix for both the local and the uploaded models. they would produce the same results, thus showing we uploaded and performed online predictions correctly.

Comment (nkashy1) [reply xgboost/notebooks/Online Prediction with XGBoost.ipynb]: @sirtorry That's a great idea.

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb (hunk 6 above)
Comment (nnegrey) [reply xgboost/notebooks/Online Prediction with XGBoost.ipynb]: Adding the steps to Install: (scikit-learn, numpy, pandas, Google API Python Client) to scikit-learn guide

File: xgboost/notebooks/Online Prediction with XGBoost.ipynb (hunk 7 above)
Comment (nnegrey) [reply xgboost/notebooks/Online Prediction with XGBoost.ipynb]: We can link them to the installation page, would that suffice?
https://cloud.google.com/sdk/downloads

Comment (nkashy1) [reply xgboost/notebooks/Online Prediction with XGBoost.ipynb]: No, linking isn't enough. I think we need to have a (very) short explanation that `gcloud` should be available on the same machine that they run their Jupyter notebook server on.

Comment (sirtorry) [reply xgboost/notebooks/Online Prediction with XGBoost.ipynb]: Note: If you get an error below, make sure the Cloud SDK is installed in the kernel's environment.

^how's that

Comment (nkashy1) [reply xgboost/notebooks/Online Prediction with XGBoost.ipynb]: I prefer something like replacing:

> [Cloud SDK](https://cloud.google.com/sdk/) is a set of tools for Cloud Platform. It contains gcloud, gsutil, and bq, which you can use to access Google Compute Engine, Google Cloud Storage, Google BigQuery, and other products and services from the command-line.

with

> [Cloud SDK](https://cloud.google.com/sdk/) is a command line tool which allows you to interact with Google Cloud products. In order to run this notebook, make sure that Cloud SDK is [installed](https://cloud.google.com/sdk/downloads) in the same environment as your Jupyter kernel.
